clang/lib/CodeGen/ModuleBuilder.cpp

```diff
@@ -180,7 +180,7 @@ namespace {
 
     bool HandleTopLevelDecl(DeclGroupRef DG) override {
       // FIXME: Why not return false and abort parsing?
-      if (Diags.hasErrorOccurred())
+      if (Diags.hasUnrecoverableErrorOccurred())
```

Comment:
Is this really a reasonable change? Doesn't that mean will be codegen'ing on erroneous code?

Reply:
I think this is reasonable, it allows codegen to continue if all of the errors which have occurred are downgradable, which happens before this change if the correct `-Wno-whatever` or `-Wno-error=whatever` option is used. We still bail out here if any of the errors are unrecoverable.